tlsutil/cert.go

```diff
@@ -68,9 +102,22 @@ func ToCertificates(data []byte) ([]Certificate, error) {
 					Organization: c.Subject.Organization,
 					CommonName:   c.Subject.CommonName,
 				},
-				DNSNames: c.DNSNames,
+				DNSNames:    c.DNSNames,
+				IPAddresses: c.IPAddresses,
 			},
 		)
 	}
 	return certificates, nil
 }
+
+// returns certificate that matches subject CN match regex (Subject.CommonName), if the certificate cannot be found,
+// second returned value will be false
+func GetCertificate(certificates []Certificate, subjectCNMatch string) (cert Certificate, ok bool) {
```

Comment:
I prefer `GetCertificateBySubjectCommonNamePattern`, or even nicer `certificates.GetBySubjectCommonNamePattern` w/ `type Certificates []Certificates`!

Reply:
like the Certificates type, what do you think of `certificates.GetBySubjectCNPattern` a bit shorter

Reply:
`certificates` in this context was a variable rather than pkg name.

So overall, I am thinking about naming the package `tlscerts` and introduce:

- `type Certificates []Certificate`
- `func (c *Certificates) GetBySubjectCommonNamePattern(...) ...`
- Include these in a newly introduced pkg `tlscerts`
- According to https://github.com/kubernetes-incubator/kube-aws/pull/1299#discussion_r187797767, you use it like:

```
certs, err := tlscerts.FromBytes(raw.content)
cert, ok := certs.GetBySubjectCommonNamePattern(...)
// do something useful with the found cert
```

WDYT?

Reply:
done, let me know what you think